mesonbuild/coredata.py

```diff
@@ -323,6 +330,7 @@ def get_builtin_option_default(optname):
     'layout':          [UserComboOption, 'Build directory layout.', ['mirror', 'flat'], 'mirror'],
     'default_library': [UserComboOption, 'Default library type.', ['shared', 'static'], 'shared'],
     'backend':         [UserComboOption, 'Backend to use.', backendlist, 'ninja'],
+    'components_default_value':  [ComponentOption, 'Value to use by default for component options.', 'auto'],
```

Comment:
Well, it is exactly what id does, you do `components_default_value=['enabled', 'disabled']` and it will make sure that all components that default to `auto` will default to that value.